src/app/rosetta/test-agent/signer.ml

```diff
@@ -32,23 +33,25 @@ let sign ~(keys : Keys.t) ~unsigned_transaction_string =
     try return (Yojson.Safe.from_string unsigned_transaction_string)
     with _ -> Result.fail (Errors.create (`Json_parse None))
   in
-  let%bind unsigned_transaction =
+  let%map unsigned_transaction =
     Transaction.Unsigned.Rendered.of_yojson json
     |> Result.map_error ~f:(fun e -> Errors.create (`Json_parse (Some e)))
     |> Result.bind ~f:Transaction.Unsigned.of_rendered
   in
+  let user_command_payload =
+    User_command_info.Partial.to_user_command_payload
+      ~nonce:unsigned_transaction.nonce
+      unsigned_transaction.Transaction.Unsigned.command
+    |> Result.ok |> Option.value_exn
+  in
   let signature =
     Schnorr.sign keys.keypair.private_key
       unsigned_transaction.random_oracle_input
+    |> Signature.Raw.encode
   in
-  let encoded_signature = Signature.Raw.encode signature in
-  let signed_transaction =
-    { Transaction.Signed.nonce= unsigned_transaction.nonce
-    ; signature= encoded_signature
-    ; command= unsigned_transaction.command }
-  in
-  let%map rendered_signed_transaction =
-    Transaction.Signed.render signed_transaction
+  let signature' =
+    User_command.sign_payload keys.keypair.private_key user_command_payload
+    |> Signature.Raw.encode
   in
-  Transaction.Signed.Rendered.to_yojson rendered_signed_transaction
-  |> Yojson.Safe.to_string
+  [%test_eq: string] signature signature' ;
```

Comment:
maybe it would be better to compare the signatures at type `Signature.t`, then you only have to encode the one you're returning

Reply:
I tried this originally but it's a bit complicated because we need to derive sexp and compare on too many things.

Reply:
Maybe you can follow up and fix this in another PR if you can see an easy way to do so